setup.sh

```diff
@@ -0,0 +1,559 @@
+    command -v triton >/dev/null 2>&1 || {
+        echo
+        echo 'Error! Joyent Triton CLI is not installed!'
+        echo 'See https://www.joyent.com/blog/introducing-the-triton-command-line-tool'
+        exit 1
+    }
+    fi
+    command -v gpg >/dev/null 2>&1 || {
+        echo
+        echo 'Error! GPG is not installed!'
+        exit 1
+    }
+
+    if [ ${COMPOSE_FILE} != "local-compose.yml" ]; then
+        # make sure Docker client is pointed to the same place as the Triton client
+        local docker_user=$(docker info 2>&1 | awk -F": " '/SDCAccount:/{print $2}')
+        local docker_dc=$(echo $DOCKER_HOST | awk -F"/" '{print $3}' | awk -F'.' '{print $1}')
+        TRITON_USER=$(triton profile get | awk -F": " '/account:/{print $2}')
+        TRITON_DC=$(triton profile get | awk -F"/" '/url:/{print $3}' | awk -F'.' '{print $1}')
+        TRITON_ACCOUNT=$(triton account get | awk -F": " '/id:/{print $2}')
+        if [ ! "$docker_user" = "$TRITON_USER" ] || [ ! "$docker_dc" = "$TRITON_DC" ]; then
+            echo
+            echo 'Error! The Triton CLI configuration does not match the Docker CLI configuration.'
+            echo "Docker user: ${docker_user}"
+            echo "Triton user: ${TRITON_USER}"
+            echo "Docker data center: ${docker_dc}"
+            echo "Triton data center: ${TRITON_DC}"
+            exit 1
+        fi
+
+        local triton_cns_enabled=$(triton account get | awk -F": " '/cns/{print $2}')
+        if [ ! "true" == "$triton_cns_enabled" ]; then
+            echo
+            echo 'Error! Triton CNS is required and not enabled.'
+            exit 1
+        fi
+
+        # setup environment file
+        if [ ! -f "_env" ]; then
+            echo '# Bootstrap via Triton CNS' >> _env
+            echo CONSUL=consul.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com >> _env
+            echo VAULT=vault.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com >> _env
+            echo >> _env
+        else
+            echo 'Existing _env file found, exiting'
+            exit
+        fi
+    fi
+}
+
+check_triton() {
+    echo
+    bold '* Checking your setup...'
+    echo './setup.sh check'
+    COMPOSE_FILE=${COMPOSE_FILE} ./setup.sh check
+}
+
+up() {
+    _demo_up
+    _demo_wait_for_consul
+}
+
+_demo_up() {
+    echo
+    bold '* Standing up the Vault cluster...'
+    echo "docker-compose -f ${COMPOSE_FILE} up -d"
+    docker-compose -f "${COMPOSE_FILE}" up -d vault consul
+    echo "docker-compose -f ${COMPOSE_FILE} scale ${service}=3"
+    docker-compose -f "${COMPOSE_FILE}" scale "${service}"=3
+}
+
+_demo_secure() {
+    echo
+    bold '* Encrypting Consul gossip and RPC'
+    secure
+}
+
+_demo_wait_for_consul() {
+    echo
+    bold '* Waiting for Consul to form raft...'
+    while :
+    do
+        docker exec -it ${vault}_1 consul info | grep -q "num_peers = 2" && break
+        echo -n '.'
+        sleep 1
+    done
+    reset # something here is breaking the terminal
```

Comment:
I've found you can avoid this problem by running docker commands that don't need user input non-interactively, by simply dropping the `-it` as I have done in my version of this script:

https://github.com/leoj3n/consul-vault/blob/08fe1e0492fe871d2885f5509d93f1a2dc80f7e8/setup.sh#L415